api/progress.go

```diff
@@ -114,10 +116,37 @@ func (r progressRoutes) saveProgress(c *gin.Context) {
 		})
 		return
 	}
+
+	streamIDString := strconv.FormatUint(uint64(request.StreamID), 10)
+	if r.StreamsDao == nil {
+		logger.Error("StreamsDao is nil")
+		return
+	}
+	stream, err := r.StreamsDao.GetStreamByID(c, streamIDString)
+
+	if err == nil {
+		lastSilence := slices.MaxFunc(stream.Silences, func(silence model.Silence, other model.Silence) int {
+			return int(silence.End) - int(other.End)
+		})
+
+		if math.Abs(float64(lastSilence.End-uint(stream.Duration.Int32))) < 10 {
+			lastSilencePercent := float64(lastSilence.Start) / float64(stream.Duration.Int32)
```

Comment:
can we be extra careful of stream.Duration not being 0? I can imagine ffmpeg fucked this up for some streams